core/shell.py

```diff
@@ -73,6 +74,30 @@
   from _devbuild.gen.runtime_asdl import Proc
 
 
+def _InitInteractiveShell(sig_state, display, my_pid):
```

Comment:
I realize it's weird that SignalState has no members

But I think it's worth it because it's a "handle" for mutation of process state.  It's actually a wrapper for global state **in the kernel** !

So we should really name it SignalDisposition or something -- that's what the man page calls this global process state

Any piece of code that registers signal handler must have the  SignalDisposition instance

It's basically a style from functional programming or OOP (dependency inversion)

----

And then GlobalTrapState() is mostly separate, or it can hold the SignalDisposition as a member